### DISCUSSION TITLE
`published_at` field in CLOTributor API

### DISCUSSION THREAD

**Discussion by aliok** [cncf/clotributor#310] [General]
Sample call: https://clotributor.dev/api/issues/search?limit=1&offset=0&sort_by=most_recent

```json
[
  {
    "number": 17811,
    "title": "🐛 Bug Report: `SidebarConfigContext` not respecting config prop values",
    "url": "https://github.com/backstage/backstage/issues/17811",
    "labels": [
      "bug",
      "help wanted"
    ],
    "published_at": 1684200943,
    "kind": "bug",
    "repository": {
      "name": "backstage",
      "url": "https://github.com/backstage/backstage",
      "homepage_url": "https://backstage.io",
      "topics": [
        "infrastructure",
        "dx",
        "developer-experience",
        "developer-portal",
        "service-catalog",
        "microservices",
        "cncf",
        "backstage",
        "hacktoberfest"
      ],
      "languages": [
        "TypeScript",
        "JavaScript",
        "Handlebars",
        "CSS",
        "Smarty"
      ],
      "stars": 21907
    },
    "project": {
      "name": "backstage",
      "display_name": "Backstage",
      "logo_url": "https://raw.githubusercontent.com/cncf/artwork/master/projects/backstage/icon/color/backstage-icon-color.svg",
      "devstats_url": "https://backstage.devstats.cncf.io/",
      "accepted_at": "2020-09-08",
      "maturity": "incubating",
      "foundation": "cncf"
    },
    "_relevance": 1
  }
]
```

I am curious about `published_at`. Is that issue creation datetime?

Are there any docs available? Or, a schema/struct/class or something similar that I can take a look?

**Comment by aliok** [comment]
Relevant: what sorting options are available other than `most_recent`?

**Comment by aliok** [comment]
Also relevant: perhaps you didn't really think that somebody would come and consume CLOTributor API, but I think it is a good data source.

**Reply by tegioz** [reply]
Thanks! 🙂

**Comment by tegioz** [comment]
> I am curious about `published_at`. Is that issue creation datetime?

Yes, that field comes from: https://github.com/cncf/clotributor/blob/b8a71fd3cb5d19b56737049074dc179983fa8bd7/clotributor-tracker/src/graphql/repo_view.graphql#L20
 
> Are there any docs available? Or, a schema/struct/class or something similar that I can take a look?

Not yet, but we'd like to add some API docs like the ones we have for [Artifact Hub](https://artifacthub.io/docs/api/) 🙂

In the meantime, maybe this can be of help:

- <https://github.com/cncf/clotributor/blob/b8a71fd3cb5d19b56737049074dc179983fa8bd7/clotributor-apiserver/src/db.rs#L67-L83>
- <https://github.com/cncf/clotributor/blob/b8a71fd3cb5d19b56737049074dc179983fa8bd7/database/migrations/functions/issues/search_issues.sql#L170-L172>

Hope that helps!

**Reply by aliok** [reply]
That helps, thanks!

**Comment by tegioz** [comment]
Out of curiosity, may I ask what would you like to build with it? 😇

**Reply by aliok** [reply]
Many people reach out to me, asking for beginner tasks to start contributing to CNCF.
I use CLOTributor to find tasks, but I want to be able to take some notes for the tasks like:
- "Person ABC said they're interested in this"
- "This task is too complex"
- "This task needs multiple persons"
- etc.

I guess it would be too much to ask to have in CLOtributor :) 

So, I thought I might have a small thing that combines the information from CLOTtributor API and my notes from a JSON file. 
I am still thinking about how to accomplish this the best way: spreadsheet generator vs modified CLOTributor webapp vs lightweight CLOTributor API client. Maybe in the end I won't do any coding but simply use a spreadsheet.

**Reply by tegioz** [reply]
That sounds cool, nice!

Maybe some of those points could somehow be summarized in the form of special labels in the issues, that we could highlight in the UI and integrate with the filters. *"This task is too complex"* kind of aligns with the *difficulty*, for example. Others may require a new special label, like [the ones described in the README file](https://github.com/cncf/clotributor#labels-with-special-meaning). But this information could be really helpful for others 🙂

Once CLOTributor becomes a bit more popular and projects are aware of how to make the most of it by applying some labels to issues,  filters will be more effective and search results more accurate, and hopefully it'll be easier for contributors to find tasks that suit their interests and skills.

**Reply by aliok** [reply]
The problem is that I cannot add labels to random CNCF projects myself :) Also, I wouldn't want that anyway. Hence, I wanted to go with a custom thing where I can store my own notes.
